internal/pkg/cli/env_init.go

```diff
@@ -20,150 +21,157 @@ import (
 			opts.prog.Stop("Done!")
-			fmt.Printf("The environment %s already exists under project %s.\n", opts.EnvName, opts.ProjectName)
+			log.Infof("The environment %s already exists under project %s.\n", opts.EnvName, opts.projectName)
 			return nil
 		}
 		opts.prog.Stop("Error!")
 		return err
 	}
 	opts.prog.Stop("Done!")
 	opts.prog.Start("Deploying env...")
-	env, err := opts.deployer.WaitForEnvironmentCreation(deployEnvInput)
+	env, err := opts.envDeployer.WaitForEnvironmentCreation(deployEnvInput)
 	if err != nil {
 		opts.prog.Stop("Error!")
 		return err
 	}
-	if err := opts.manager.CreateEnvironment(env); err != nil {
+	if err := opts.envCreator.CreateEnvironment(env); err != nil {
 		opts.prog.Stop("Error!")
 		return err
 	}
 	opts.prog.Stop("Done!")
 	return nil
 }
 
+// RecommendedActions returns follow-up actions the user can take after successfully executing the command.
+func (opts *InitEnvOpts) RecommendedActions() []string {
+	return nil
+}
+
 // BuildEnvInitCmd builds the command for adding an environment.
 func BuildEnvInitCmd() *cobra.Command {
 	opts := InitEnvOpts{
 		EnvProfile: "default",
 		prog:       spinner.New(),
-		prompter:   prompt.New(),
+		prompt:     prompt.New(),
 	}
 
 	cmd := &cobra.Command{
 		Use:   "init [name]",
 		Short: "Create a new environment in your project.",
 		Example: `
-  Create a test environment in your "default" AWS profile
+  Create a test environment in your "default" AWS profile.
   /code $ archer env init test
 
-  Create a prod-iad environment using your "prod-admin" AWS profile
+  Create a prod-iad environment using your "prod-admin" AWS profile.
   /code $ archer env init prod-iad --profile prod-admin --prod`,
 		PreRunE: func(cmd *cobra.Command, args []string) error {
 			if len(args) > 0 {
 				opts.EnvName = args[0]
 			}
-
-			opts.ProjectName = viper.GetString("project")
-			// If the project flag or env name isn't passed in, ask the user for them.
-			if err := opts.Ask(); err != nil {
+			store, err := ssm.NewStore()
+			if err != nil {
 				return err
 			}
-			return nil
-		},
-		RunE: func(cmd *cobra.Command, args []string) error {
-			// TODO: pass in configured session to ssm.NewStore?
-			s, err := ssm.NewStore()
+			profileSess, err := session.FromProfile(opts.EnvProfile)
 			if err != nil {
 				return err
 			}
-			opts.manager = s
-			opts.projectGetter = s
-
-			sess, err := session.FromProfile(opts.EnvProfile)
+			defaultSession, err := session.Default()
 			if err != nil {
 				return err
 			}
-			opts.deployer = cloudformation.New(sess)
-
-			defaultSession, err := session.Default()
+			opts.envCreator = store
+			opts.projectGetter = store
+			opts.envDeployer = cloudformation.New(profileSess)
 			opts.identity = identity.New(defaultSession)
-
+			opts.projectName = viper.GetString("project")
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			log.Warningln("It's best to run this command in the root of your workspace.")
```

Comment:
I don’t think we need this warning - since adding an environment doesn’t affect workspace state.